src/averageAges.js

```diff
@@ -15,31 +15,53 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let peopleCentury = [...people];
+
+  if (century !== null && typeof century === 'number') {
+    peopleCentury = people.filter(person =>
+      Math.ceil(person.died / 100) === century);
+  }
+
+  const men = peopleCentury.filter(person => person.sex === 'm');
+
+  const sumAge = men.reduce((total, person) =>
+    total + (person.died - person.born), 0);
+  const averageAge = sumAge / men.length;
+
+  return averageAge;
 }
 
 /**
  * Implement calculateWomenAverageAge function
  *
- * Function returns average age of women in array. If `withChildren` is
+ * Function returns average age of women in array. If `withwomen` is
  * specified then function calculates average age only for women with children
  *
- * Hint: To check if a woman has children you should find someone who mention
+ * Hint: To check if a woman has women you should find someone who mention
  * her as mother.
  *
  * @param {object[]} people
- * @param {boolean} withChildren - optional
+ * @param {boolean} withwomen - optional
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+function calculateWomenAverageAge(people, withwomen) {
+  let women = [...people];
+
+  if (withwomen === true) {
+    women = people.filter(person =>
+      person.sex === 'f' && people.some(p => p.mother === person.name));
+  } else {
+    women = women.filter(person => person.sex === 'f');
+  }
+
+  const sumAge = women.reduce((total, person) =>
+    total + (person.died - person.born), 0);
 
+  const averageAge = sumAge / women.length;
```

Comment:
It would be great to create a function-helper for calculating the average age as this logic duplicates both for men and women